backend/src/main/java/saci/domain/service/KnowledgeService.java

```diff
@@ -21,17 +23,22 @@ public class KnowledgeService {
     public Knowledge createKnowledge(Knowledge knowledge) {
         Role role =
                 roleRepository
-                        .findById((knowledge.getRoleId()))
-                        .orElseThrow(() -> new NotFoundException("Role Not Found"));
+                        .findById(knowledge.getRoleId())
+                        .orElseThrow(() -> {
+                            String errorMessage = "Role Not Found";
+                            log.error(errorMessage);
+                            return new NotFoundException(errorMessage);
+                        });
+
         boolean knowledgeWithNameExists =
                 role.getKnowledges().stream()
-                        .anyMatch(
-                                existingKnowledge ->
-                                        existingKnowledge.getName().equals(knowledge.getName()));
+                        .anyMatch(existingKnowledge ->
+                                existingKnowledge.getName().equals(knowledge.getName()));
 
         if (knowledgeWithNameExists) {
-            throw new AlreadyExistsException(
-                    "Knowledge with the same name already exists in this role");
+            String errorMessage = "Knowledge with the same name already exists  ";
+            log.error(errorMessage);
```

Comment:
Instead of logging in every place, why not log inside the exception method you are throwing? This way, you'll just have one log line